src/net_processing.cpp

```diff
@@ -2154,10 +2197,24 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
             nCMPCTBLOCKVersion = 1;
             connman->PushMessage(pfrom, msgMaker.Make(NetMsgType::SENDCMPCT, fAnnounceUsingCMPCTBLOCK, nCMPCTBLOCKVersion));
         }
+
         pfrom->fSuccessfullyConnected = true;
         return true;
     }
 
+    // Feature negotiation of wtxidrelay should happen between VERSION and
+    // VERACK, to avoid relay problems from switching after a connection is up
```

Comment:
This comment is slightly confusing to me. If you setting this expectation here, maybe you want to enforce the rule in the code? Track whether VERACK was not received or something.

Reply:
What to do in that case?  We could disconnect the peer for violating the spec, or ignore the message and not switch to wtxid relay, but I thought trying to do wtxid-relay was the most practical, as it should mostly work (just things that were announced earlier might not be successfully requested).  Maybe it's better to not tolerate any kind of bugs though, so that implementations get all this stuff right?

Reply:
Disconnecting is too much, but ignoring the wtxid_relay message may be reasonable.

Alternatively (and probably better), we can just make the comment more clear. Something like "currently this is not enforced, because at most we get slight inefficiency if not following this rule".

And then the responsibility would be on the future developer to introduce the disconnect/ignore behavior if they choose to fundamentally rely on this assumption.

Reply:
Ok made it disconnect. Failing to make any connections at all is a good way for software to get fixed I guess...